config/logging.yaml

```diff
@@ -10,13 +10,13 @@
 #   Enable logging output inline in rails console
 #   values: true, false
 #
-# :level: 
+# :level:
 #   Only messages with severity >= this level from all loggers will be logged
 #   values: debug, info, warn, error, fatal, unknown
 #
 # :type:
 #   Type of logging, for file variant you can specify other options below like filename
-#   values: file, syslog
+#   values: file, syslog or journald
```

Comment:
Pardon my ignorance here, do you mind a brief reminder of what the difference between syslog and journald are?

Reply:
Sure, syslog has an "API" that only allows sending strings while system journal supports message and arbitrary amount of key=value pairs. Supporting both give options to users:

* send logs to stdout/file in plain text
* send logs to syslog in plain text
* send logs to syslog in JSON
* send logs to journal in plain text with additional fields
* send logs to journal in JSON with additional fields

If you want to integrate with logstask (ELK) or similar 3rd party, you will likely use syslog/JSON configuration. If you want to keep standalone logging, the most flexible configuration (I am going to propose as sane default) will be journal/plain text. Advantages:

* messages are human-readable and it is approachable by users
* log entries are enriched with extra fields
* by default journald forwards to rsyslog on EL7 so extra data is lost
* user can decide to disable volatile journal and start capturing logs with extra fields
* thank to syslog facility setting, installer will configure rsyslog to send those into production.log (no big change for users - everything stays "as-is")
* journald can be easily configured to pull logs from remote hosts in the same format (smart proxies - the next step in my effort)
* with very little configuration changes (disable volatile, configure fetching from proxies) user can achive good central structured logging with extra features like log sealing and filtering

Hope it makes more sense now. The key is ability to integrate with both worlds.